gtdynamics/universal_robot/sdf.cpp

```diff
@@ -75,19 +75,26 @@ JointParams ParametersFromSdfJoint(const sdf::Joint &sdf_joint) {
 Pose3 GetJointFrame(const sdf::Joint &sdf_joint,
                     const LinkSharedPtr &parent_link,
                     const LinkSharedPtr &child_link) {
-  auto frame = sdf_joint.PoseFrame();
-  if (frame == "" || frame == child_link->name()) {
-    if (sdf_joint.Pose() == ignition::math::Pose3d())
-      return child_link->wTl();
-    else
-      return child_link->wTl() * Pose3FromIgnition(sdf_joint.Pose());
-  } else if (frame == parent_link->name()) {
-    if (sdf_joint.Pose() == ignition::math::Pose3d())
-      return parent_link->wTl();
-    else
-      return parent_link->wTl() * Pose3FromIgnition(sdf_joint.Pose());
-  } else if (frame == "world") {
-    return Pose3FromIgnition(sdf_joint.Pose());
+  // Name of the coordinate frame the joint's pose is relative to.
+  // Specified by `relative_to` in the SDF file.
+  std::string frame_name = sdf_joint.PoseRelativeTo();
+
+  // Get the pose of the joint in the parent or child link's frame depending on
+  // the value of `frame_name`.
+  Pose3 lTj = Pose3FromIgnition(sdf_joint.RawPose());
```

Comment:
But, it's not lTj. We don't know that yet... Re-read my comments? :-)

Reply:
The SDF spec mentions that all joint raw poses are with respect to the parent (or in some cases, the child) link's frame. This depends on what `relative_to` is set to for the joint. Here are the [docs](http://osrf-distributions.s3.amazonaws.com/sdformat/api/10.3.0/classsdf_1_1v10_1_1Joint.html#a69930bc6f744b43448db88d9255d4b1c).

Reply:
I understand. So whether it is `lTj` is not known yet, hence my comment.

Reply:
But it is known. It's known from the definition of the SDF file.

Reply:
Just need this last one resolved. Should I rename it to `raw_pose`?